Remove aoti_torch_cpu__weight_int4pack_mm_cpu_tensor

janeyx99 opened the pull request:
I noticed that this op was likely intended to be in the `extern "C"` portion of the file, but it was not added as such in https://github.com/pytorch/pytorch/pull/145250 which means this function is actually not stable/would get mangled by C++.

Following the thread there I am thinking there are two possible solutions:
(1) Since this op was never stable to begin with, and @Xia-Weiwen should add it to fallback_ops.py, I think this op is deletable + should get deleted before the 2.7 branch cut. 
(2) Or we could just move the op to the right portion of the code. ~While I like just deleting the op, I am hesitant to do in case there's something I haven't considered, so this PR does option 2.~

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #148834

janeyx99 committed:
Move aoti_torch_cpu__weight_int4pack_mm_cpu_tensor to not be mangled

[ghstack-poisoned]

desertfire commented:
Did you try my suggestion here, https://github.com/pytorch/pytorch/pull/145250#discussion_r1954568031 ?

janeyx99 commented:
> Did you try my suggestion here, [#145250 (comment)](https://github.com/pytorch/pytorch/pull/145250#discussion_r1954568031) ?

@Xia-Weiwen said they’d do it in https://github.com/pytorch/pytorch/pull/146756 but I’m not sure it actually got done

desertfire commented:
> > Did you try my suggestion here, [#145250 (comment)](https://github.com/pytorch/pytorch/pull/145250#discussion_r1954568031) ?
> 
> @Xia-Weiwen said they’d do it in #146756 but I’m not sure it actually got done

I don't think they did. Since you are already looking at it, do you want to change this PR to do that?

desertfire commented:
Landing now to catch the branch cut. We can improve it as a follow-up.

desertfire commented:
@pytorchbot merge

desertfire commented:
@pytorchbot merge -i

Xia-Weiwen commented:
> > > Did you try my suggestion here, [#145250 (comment)](https://github.com/pytorch/pytorch/pull/145250#discussion_r1954568031) ?
> > 
> > 
> > @Xia-Weiwen said they’d do it in #146756 but I’m not sure it actually got done
> 
> I don't think they did. Since you are already looking at it, do you want to change this PR to do that?

Sorry I forgot to do that. Do I still need to do that? Thanks.

desertfire commented:
> Sorry I forgot to do that. Do I still need to do that? Thanks.

Please!

janeyx99 commented:
@pytorchbot revert -c weird -m "sorry I don't think I want this PR in before the branch cut, as it'd freeze the API in the file when it should really be in a different header"

janeyx99 committed:
Update on "Move aoti_torch_cpu__weight_int4pack_mm_cpu_tensor to not be mangled"

I noticed that this op was likely intended to be in the `extern "C"` portion of the file, but it was not added as such in https://github.com/pytorch/pytorch/pull/145250 which means this function is actually not stable/would get mangled by C++.

Following the thread there I am thinking there are two possible solutions:
(1) Since this op was never stable to begin with, and Xia-Weiwen already landed the fallback, maybe this op is deletable + should get deleted before the 2.7 branch cut
(2) Or we could just move the op to the right portion of the code. While I like just deleting the op, I am hesitant to do in case there's something I haven't considered, so this PR does option 2.




[ghstack-poisoned]

janeyx99 committed:
Update on "Move aoti_torch_cpu__weight_int4pack_mm_cpu_tensor"

I noticed that this op was likely intended to be in the `extern "C"` portion of the file, but it was not added as such in https://github.com/pytorch/pytorch/pull/145250 which means this function is actually not stable/would get mangled by C++.

Following the thread there I am thinking there are two possible solutions:
(1) Since this op was never stable to begin with, and Xia-Weiwen already landed the fallback, maybe this op is deletable + should get deleted before the 2.7 branch cut
(2) Or we could just move the op to the right portion of the code. While I like just deleting the op, I am hesitant to do in case there's something I haven't considered, so this PR does option 2.




[ghstack-poisoned]

janeyx99 commented:
Closing as this has been completely resolved
